File: Easy/2108. Find First Palindromic String in the Array/Solution.java
class Solution {
    public String firstPalindrome(String[] words) {
        for (String string : words) {
            if (checkPal(string)) {
                return string;
            }
        }
        return "";
    }

    private boolean checkPal(String string) {
        int i = 0, j = string.length() - 1;

        while (i < j) {
            if (string.charAt(i) != string.charAt(j)) {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}

File: Easy/2108. Find First Palindromic String in the Array/Solution.cpp
#include <string>
#include <vector>

class Solution
{
public:
    std::string firstPalindrome(std::vector<std::string> &words)
    {
        for (const std::string &str : words)
        {
            if (chekPal(str))
            {
                return str;
            }
        }

        return "";
    }

private:
    bool chekPal(const std ::string &str)
    {
        int leftP = 0, rightP = str.length() - 1;

        while (leftP < rightP)
        {
            if (str[leftP] != str[rightP])
            {
                return false;
            }
            leftP++;
            rightP--;
        }

        return true;
    }
};